resources/main.mjs

```diff
@@ -0,0 +1,273 @@
+        gaugeNeedleElement.style.setProperty('-webkit-transform', needleRotationValue);
+        gaugeNeedleElement.style.setProperty('-moz-transform', needleRotationValue);
+        gaugeNeedleElement.style.setProperty('-ms-transform', needleRotationValue);
+        gaugeNeedleElement.style.setProperty('transform', needleRotationValue);
+    }
+
+    _populateDetailedResults(formattedValues) {
+        const resultsTables = document.querySelectorAll('.results-table');
+        let i = 0;
+        resultsTables[0].innerHTML = '';
+        for (; i < Math.ceil(formattedValues.length / 2); i++)
+            this._addDetailedResultsRow(resultsTables[0], i, formattedValues[i]);
+        resultsTables[1].innerHTML = '';
+        for (; i < formattedValues.length; i++)
+            this._addDetailedResultsRow(resultsTables[1], i, formattedValues[i]);
+    }
+
+    prepareUI() {
+        window.addEventListener('popstate', this._popStateHandler.bind(this), false);
+        window.addEventListener('resize', this._resizeScreeHandler.bind(this));
+        this._resizeScreeHandler();
+
+        document.getElementById("logo").onclick = this._logoClickHandler.bind(this);
+        document.getElementById("show-summary").onclick = (e) => this.showResultsSummary();
+        document.getElementById("show-details").onclick = (e) => this.showResultsDetails();
+        document.querySelectorAll(".show-about button").forEach(
+            each => { each.onclick = () => this._showSection('about', true); }
+        );
+        document.querySelectorAll(".start-tests-button").forEach(
+            button => { button.onclick = this._startBenchmarkHandler.bind(this); }
+        );
+    }
+
+    _popStateHandler(event) {
+        if (event.state) {
+            const sectionToShow = event.state.section;
+            if (sectionToShow) {
+                const sections = document.querySelectorAll('main > section');
+                for (let i = 0; i < sections.length; i++) {
+                    if (sections[i].id === sectionToShow)
+                        return this._showSection(sectionToShow, false);
+                }
+            }
+        }
+        return this._showSection('home', false);
+    }
+
+    _resizeScreeHandler() {
+        // FIXME: Detect when the window size changes during the test.
+        const screenIsTooSmall = window.innerWidth < 850 || window.innerHeight < 650;
+        document.getElementById('screen-size').textContent = window.innerWidth + 'px by ' + window.innerHeight + 'px';
+        document.getElementById('screen-size-warning').style.display = screenIsTooSmall ? null : 'none';
+    }
+
+    _startBenchmarkHandler() {
+        if (this.startBenchmark())
+            this._showSection('running');
+    }
+
+    _logoClickHandler(event) {
+        // Prevent any accidental UI changes during benchmark runs.
+        if (!this._isRunning) this._showSection('home', true);
```

Comment:
Please insert a line break between `)` and `this`.

Reply:
DONE.